packages/base-types/src/models/apiKey.ts

```diff
@@ -27,4 +28,5 @@ export interface Model<Data extends AnyRecord = AnyRecord> {
   data?: Data;
   scopes: string[];
   permissions: string[];
+  secondaryKeyID: string;
```

Comment:
Instead of the parent referencing the child, we could flip this and have the secondary key reference the primary key.

```suggestion
  parentKeyID: string;
```

That way we can identify a primary key (`parentKeyID === null`) and also a secondary key (`parentKeyID !== null`) without requiring a new `APIKeySubType`.